File: src/main/java/org/example/eventhub/dto/order/OrderResponseLong.java
package org.example.eventhub.dto.order;

import java.math.BigDecimal;
import java.util.List;
import org.example.eventhub.dto.ticket.TicketResponseShort;
import org.example.eventhub.dto.user.UserResponseShort;
import org.example.eventhub.enums.OrderStatus;

public record OrderResponseLong(
        Long id,
        UserResponseShort user,
        List<TicketResponseShort> tickets,
        BigDecimal totalPrice,
        OrderStatus orderStatus) {}

File: src/main/java/org/example/eventhub/dto/order/OrderResponseLong.kt
package org.example.eventhub.dto.order

import org.example.eventhub.dto.ticket.TicketResponseShort
import org.example.eventhub.dto.user.UserResponseShort
import org.example.eventhub.enums.OrderStatus
import java.math.BigDecimal

data class OrderResponseLong(
    val id: Long?,
    val user: UserResponseShort,
    val tickets: List<TicketResponseShort>,
    val totalPrice: BigDecimal,
    val orderStatus: OrderStatus
)
